src/components/KeyboardCompatibleView/KeyboardCompatibleView.js

```diff
@@ -20,63 +25,96 @@ import { KeyboardContext } from '../../context';
  * </KeyboardCompatibleView>
  * ```
  */
-export const KeyboardCompatibleView = ({
-  children,
-  enabled = true,
-  keyboardDismissAnimationDuration = 500,
-  keyboardOpenAnimationDuration = 500,
-}) => {
-  const heightAnim = useRef(new Animated.Value(0)).current;
-  const rootChannelView = useRef();
-
-  const [initialHeight, setInitialHeight] = useState(0);
-
-  const { height: channelHeight, isKeyboardOpen } = useKeyboardCompatibleHeight(
-    {
-      enabled,
-      initialHeight,
-      rootChannelView,
-    },
-  );
+export const KeyboardCompatibleView = ({ children, enabled = true }) => {
+  const appState = useAppState();
+  const [channelHeight, setChannelHeight] = useState(0);
+
+  const rootChannelView = useRef(null);
+  const isKeyboardOpen = useRef(false);
+  const initialHeight = useRef(0);
 
   useEffect(() => {
-    Animated.timing(heightAnim, {
-      duration: isKeyboardOpen
-        ? keyboardDismissAnimationDuration
-        : keyboardOpenAnimationDuration,
-      toValue: channelHeight,
-      useNativeDriver: false,
-    }).start();
-  }, [
-    channelHeight,
-    heightAnim,
-    keyboardDismissAnimationDuration,
-    keyboardOpenAnimationDuration,
-  ]);
-
-  const dismissKeyboard = useCallback(() => {
-    Keyboard.dismiss();
-
-    return new Promise((resolve) => {
-      if (!isKeyboardOpen) {
-        // If channel height is already at full length, then don't do anything.
-        resolve();
+    const onKeyboardChange = (e) => {
+      if (!enabled) {
+        return;
+      }
+
+      if (e === null) {
+        setChannelHeight(initialHeight.current);
+        isKeyboardOpen.current = false;
+        return;
+      }
+
+      const { endCoordinates } = e;
+      let isOpening = true;
+
+      if (Platform.OS === 'ios') {
+        const { startCoordinates } = e;
+        isOpening = endCoordinates.screenY < startCoordinates.screenY;
+      }
+
+      const keyboardY = endCoordinates.screenY;
+      if (rootChannelView) {
+        if (e.duration && e.easing) {
+          LayoutAnimation.configureNext({
+            // We have to pass the duration equal to minimal accepted duration defined here: RCTLayoutAnimation.m
+            duration: e.duration > 10 ? e.duration : 10,
+            update: {
+              duration: e.duration > 10 ? e.duration : 10,
+              type: LayoutAnimation.Types[e.easing] || 'keyboard',
+            },
+          });
+        }
+        rootChannelView.current.measureInWindow((x, y) => {
+          if (Platform.OS === 'android') {
+            setChannelHeight(
+              isOpening
```

Comment:
based on the logic earlier, seems like `isOpening` will always be true for the android platform as it's only changed within an if statement that's iOS specific